node/src/components/small_network.rs

```diff
@@ -182,9 +182,15 @@ where
     /// era, so we treat the highest era we have seen as the active era.
     highest_era_seen: EraId,
 
-    /// The active bandwidth limiter.
+    /// The outgoing bandwidth limiter.
     #[data_size(skip)]
-    limiter: Box<dyn BandwidthLimiter>,
+    upstream_limiter: Box<dyn Limiter>,
```

Comment:
Why `Box<dyn Limiter>` and not something like `L1: Limiter, L2: Limiter`?

Reply:
> My own preference would be to avoid the `Limiter` trait altogether and change the `ClassBasedLimiter` to handle the single special case of being constructed with resources_per_second == 0, implying unlimited resources.

I wanted to suggest the same :+1:.

Reply:
> I believe it's so we can choose at runtime whether to hold a ClassBasedLimiter or an Unlimited one.

That's correct. Otherwise we'd only be able to switch them at compile time.

Otherwise I disagree, we may want to try additional strategies for limiting in the future (fixed rate per peer, HTBs, etc.) and this neatly captures it. Every one of these would need a special case for `== 0` or similar.

The alternative I can see is to use an `Option`, but that seemed less elegant.

Reply:
I'm fine with the answer, resolving the comment Fraser. If you still want to continue a discussion you can unresolve it.